### PR TITLE
Deploy docs CFN Fixes

### PR DESCRIPTION
**Pull request by drworm**
* Auto assign public IP without having to make changes on the VPC
* Fixed the NetworkInterface reference

### CODE REVIEW

File: docs/deploying.md [hunk 1]
```diff
@@ -167,15 +166,24 @@ Resources:
...
+          Fn::Base64:
+            Fn::Join:
+              - ""
+              - - |-
+                  #!/bin/bash
+                  echo "eni_id=
+                - !Ref FckNatInterface
+                - |-
+                  " >> /etc/fck-nat.conf
+                  service fck-nat restart
```
Comment (AndrewGuenther) [inline docs/deploying.md:186]: I don't think there were any issues with the previous version of this? I would much prefer to keep this using `!Sub`, it's a lot cleaner.

File: docs/deploying.md [hunk 2]
```diff
@@ -167,15 +166,24 @@ Resources:
       LaunchTemplateData:
         ImageId: !Ref FckNatAMIParameter
         InstanceType: t4g.nano
+        NetworkInterfaces:
+          - DeviceIndex: 0
+            AssociatePublicIpAddress: true
+            Groups:
+            - !GetAtt [FckNatSecurityGroup, GroupId]
```
Comment (AndrewGuenther) [inline docs/deploying.md:173]: While this is more explicit, if you are deploying your NAT into a public subnet, associating a public IP address is the default.

**Review by AndrewGuenther — COMMENTED** [review]
I

File: docs/deploying.md (hunk 1 above)
Comment (drworm) [reply docs/deploying.md]: Yeah sorry about that, I've reverted this change

File: docs/deploying.md (hunk 2 above)
Comment (drworm) [reply docs/deploying.md:173]: I believe by default, and EC2 instance launched in a public subnet is not assigned a public IP.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-subnet.html#cfn-ec2-subnet-mappubliciponlaunch

This is just a problem I ran into, and I didn't want to go changing my VPC defaults just to launch a NAT instance

**Review by AndrewGuenther — APPROVED** [review]

**Pull request merged**